src/Builder/Visitor/AddMethodCallToCallListVisitor.php

```diff
@@ -0,0 +1,197 @@
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\Builder\Visitor;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\NodeFinder;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitorAbstract;
+use SprykerSdk\Integrator\Transfer\ClassMetadataTransfer;
+
+class AddMethodCallToCallListVisitor extends NodeVisitorAbstract
+{
+    /**
+     * @var string
+     */
+    protected const ARRAY_MERGE_FUNCTION = 'array_merge';
+
+    /**
+     * @var string
+     */
+    protected const THIS_KEY = 'this';
+
+    /**
+     * @var \SprykerSdk\Integrator\Transfer\ClassMetadataTransfer
+     */
+    protected $classMetadataTransfer;
+
+    /**
+     * @var bool
+     */
+    protected $methodFound = false;
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassMetadataTransfer $classMetadataTransfer
+     */
+    public function __construct(ClassMetadataTransfer $classMetadataTransfer)
+    {
+        $this->classMetadataTransfer = $classMetadataTransfer;
+    }
+
+    /**
+     * @param \PhpParser\Node $node
+     *
+     * @return \PhpParser\Node|int
+     */
+    public function enterNode(Node $node)
+    {
+        $callMetadataTransfer = $this->classMetadataTransfer->getCall();
+        if (!$callMetadataTransfer) {
+            return $node;
+        }
+
+        $callTargetMethodName = $this->getMethodNameFromNamespaceName($callMetadataTransfer->getTargetOrFail());
+        if ($node instanceof ClassMethod && $node->name->toString() === $callTargetMethodName) {
+            $this->methodFound = true;
+
+            return $node;
+        }
+
+        if ($this->methodFound) {
+            if ($node instanceof FuncCall && $this->isArrayMergeFuncCallNode($node)) {
+                $this->addNewCallIntoArrayMergeFuncNode($node);
+
+                return $this->successfullyProcessed();
+            }
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\FuncCall $node
+     *
+     * @return bool
+     */
+    protected function isArrayMergeFuncCallNode(FuncCall $node): bool
+    {
+        return $node->name instanceof Name && $node->name->parts[0] === static::ARRAY_MERGE_FUNCTION;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\FuncCall $node
+     *
+     * @return \PhpParser\Node
+     */
+    protected function addNewCallIntoArrayMergeFuncNode(FuncCall $node): Node
+    {
+        $calledMethods = $this->getCalledMethods($node);
+        $callMetadataTransfer = $this->classMetadataTransfer->getCall();
+        if (!$callMetadataTransfer) {
+            return $node;
+        }
+
+        $before = $callMetadataTransfer->getBefore();
+        if ($before) {
+            $beforeMethodName = $this->getMethodNameFromNamespaceName($before);
```

Comment:
Instead of `getMethodNameFromNamespaceName` i'd create method like `getPositionByNamespace($calledMethods, $needleNamespace): ?int` deu to before and after has the same logic

Reply:
`getMethodNameFromNamespaceName` used in the `enterNode` method too 

I added `getPositionByNamespace` in addition to the `getMethodNameFromNamespaceName`